romancal/outlier_detection/outlier_detection_step.py

```diff
@@ -87,81 +113,36 @@ def process(self, input_models):
-
-        if not self.valid_input:
-            if self.input_container:
-                for model in self.input_models:
-                    model.meta.cal_step.outlier_detection = "SKIPPED"
-            else:
-                self.input_models.meta.cal_step.outlier_detection = "SKIPPED"
-            self.skip = True
-            return self.input_models
-
         self.log.debug(f"Using {detection_step.__name__} class for outlier_detection")
 
         # Set up outlier detection, then do detection
         step = detection_step(self.input_models, **pars)
         step.do_detection()
 
         state = "COMPLETE"
-        if self.input_container:
+
+        if not self.save_intermediate_results:
+            self.log.debug(
+                "The following files will be deleted since save_intermediate_results=False:"
+            )
+        for model in self.input_models:
+            model.meta.cal_step.outlier_detection = state
             if not self.save_intermediate_results:
-                self.log.debug(
-                    "The following files will be deleted since save_intermediate_results=False:"
-                )
-            for model in self.input_models:
-                model.meta.cal_step.outlier_detection = state
-                if not self.save_intermediate_results:
-                    #  Remove unwanted files
-                    crf_path = self.make_output_path(basepath=model.meta.filename)
-                    #  These lines to be used when/if outlier_i2d files follow output_dir
-                    #  crf_file = os.path.basename(crf_path)
-                    #  outlr_path = crf_path.replace(crf_file, outlr_file)
-                    outlr_file = model.meta.filename.replace("cal", "outlier_i2d")
-                    blot_path = crf_path.replace("crf", "blot")
-                    median_path = blot_path.replace("blot", "median")
-
-                    for fle in [outlr_file, blot_path, median_path]:
-                        if os.path.isfile(fle):
-                            os.remove(fle)
-                            self.log.debug(f"    {fle}")
-        else:
-            self.input_models.meta.cal_step.outlier_detection = state
-        return self.input_models
+                #  Remove unwanted files
+                crf_path = self.make_output_path(basepath=model.meta.filename)
+                #  These lines to be used when/if outlier_i2d files follow output_dir
+                #  crf_file = os.path.basename(crf_path)
+                #  outlr_path = crf_path.replace(crf_file, outlr_file)
+                outlr_file = model.meta.filename.replace("cal", "outlier_i2d")
+                blot_path = crf_path.replace("crf", "blot")
+                median_path = blot_path.replace("blot", "median")
+
+                for fle in [outlr_file, blot_path, median_path]:
+                    if os.path.isfile(fle):
+                        os.remove(fle)
+                        self.log.debug(f"    {fle}")
```

Comment:
Does it make sense to put the code dealing with cleaning of generated files in a separate function?